core/services/relay/evm/size_helper.go

```diff
@@ -0,0 +1,72 @@
+package evm
+
+import (
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+func GetMaxSizeFormEntry(n int, entry *CodecEntry) (int, error) {
+	if entry == nil {
+		return 0, commontypes.ErrInvalidType
+	}
+	return GetMaxSize(n, entry.Args)
+}
+
+func GetMaxSize(n int, args abi.Arguments) (int, error) {
+	size := 0
+	for _, arg := range args {
+		tmp := arg.Type
+		argSize, _, err := getTypeSize(n, &tmp, true, false)
+		if err != nil {
+			return 0, err
+		}
+		size += argSize
+	}
+
+	return size, nil
+}
+
+func getTypeSize(n int, t *abi.Type, dynamicTypeAllowed bool, isNested bool) (int, bool, error) {
```

Comment:
nit, why is abi.Type a pointer?

Reply:
wdym, why would you copy it by value. If you pass a dereferenced `arg.Type` into ` getTypeSize` you get the same effect of passing &tmp. Non pointer attributes inside of the struct won't get changed and attributes with a pointer would get changed either way.

Reply:
See below, t.Elm is passed in on line 27, it's naturally a pointer.  When you pass a struct directly, you force a copy of that struct to be made.